### PR TITLE
Add health check mixin

### PR DESCRIPTION
**Pull request by Americas**

### CODE REVIEW

File: test/acceptance/health-check.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            const healthCheck = reader.getHealthCheck();
+
+            assert.equal(healthCheck.connections.active, 1);
+            assert.equal(healthCheck.connections.expected, 1);
+
+            reader.end();
+
+            done();
+          } catch (error) {
+            done(error);
```
Comment (jspranger) [inline test/acceptance/health-check.js:48]: I don't understand the need to invoke `done()` for both success and fail. Feel like it misses the point of actually testing a success of failure case.

Comment (Americas) [reply test/acceptance/health-check.js:48]: Invoking `done()` with an argument is basically saying that the test failed. If `done()` is not called, then the test will keep running until it times out.

Comment (jspranger) [reply test/acceptance/health-check.js:48]: Not very fond of replicating the pattern, but it is what it is.

File: lib/mixins/health-check.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @param {Reader|Writer} client
+ * @api private
+ */
+
+module.exports = client => {
+  if (!client.healthCheck) {
+    return;
+  }
+
+  client.getHealthCheck = () => {
```
Comment (jspranger) [inline lib/mixins/health-check.js]:
```suggestion
  client.getHealthStatus = () => {
```
?

Comment (Americas) [reply lib/mixins/health-check.js]: Updated.

**Review by jspranger — APPROVED** [review]
Tested in `development-reliability`.

LGTM.

**Pull request merged**